packages/dev-server/src/HermesBundler.ts

```diff
@@ -0,0 +1,154 @@
+    if (!fs.existsSync(gradlePropertiesPath)) {
+      // TODO(kudo): Clarify default behavior in managed workflow for gradle.properties not exist
+      return false;
+    }
+    const properties = parseGradleProperties(await fs.readFile(gradlePropertiesPath, 'utf8'));
+    for (const [key, value] of properties) {
+      if (key === 'JS_RUNTIME' && value === 'hermes') {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+interface HermesBundleOutput {
+  hbc: Uint8Array;
+  sourcemap: string;
+}
+export async function buildHermesBundleAsync(
+  projectRoot: string,
+  code: string,
+  map: string,
+  optimize: boolean = false
+): Promise<HermesBundleOutput> {
+  const tempDir = path.join(os.tmpdir(), `expo-bundler-${process.pid}`);
+  await fs.ensureDir(tempDir);
+
+  const tempBundleFile = path.join(tempDir, 'index.bundle');
+  const tempSourcemapFile = path.join(tempDir, 'index.bundle.map');
+  await fs.writeFile(tempBundleFile, code);
+  await fs.writeFile(tempSourcemapFile, map);
+
+  const tempHbcFile = path.join(tempDir, 'index.hbc');
+  const hermesCommand = getHermesCommand(projectRoot);
+  const args = ['-emit-binary', '-out', tempHbcFile, tempBundleFile, '-output-source-map'];
+  if (optimize) {
+    args.push('-O');
+  }
+  await spawnAsync(hermesCommand, args);
+  const hbc = await fs.readFile(tempHbcFile);
+
+  const sourcemap = await createHermesSourcemapAsync(projectRoot, map, `${tempHbcFile}.map`);
+
+  await fs.remove(tempDir);
+
+  return {
+    hbc,
+    sourcemap,
+  };
+}
+
+export async function createHermesSourcemapAsync(
+  projectRoot: string,
+  sourcemap: string,
+  hermesMapFile: string
+): Promise<string> {
+  const composeSourceMaps = importMetroSourceMapFromProject(projectRoot);
+  const bundlerSourcemap = JSON.parse(sourcemap);
+  const hermesSourcemap = await fs.readJSON(hermesMapFile);
+  return JSON.stringify(composeSourceMaps([bundlerSourcemap, hermesSourcemap]));
+}
+
+function getHermesCommand(projectRoot: string): string {
+  const platform = getHermesCommandPlatform();
+  const resolvedPath = resolveFrom.silent(projectRoot, `hermes-engine/${platform}/hermesc`);
+  if (!resolvedPath) {
+    throw new Error(
+      `Missing module "hermes-engine/${platform}/hermesc" in the project.` +
+        'This usually means hermes-engine is not installed. ' +
+        'Please verify that dependencies in package.json include "react-native" ' +
+        'and run `yarn` or `npm install`.'
+    );
+  }
+  return resolvedPath;
+}
+
+function getHermesCommandPlatform(): string {
+  switch (os.platform()) {
+    case 'darwin':
+      return 'osx-bin';
+    case 'linux':
+      return 'linux64-bin';
+    case 'win32':
+      return 'win64-bin';
+    default:
+      throw new Error(`Unsupported host platform for Hermes compiler: ${os.platform()}`);
+  }
+}
+
+export function parseGradleProperties(content: string): [string, string][] {
```

Comment:
is this the same thing as https://github.com/expo/expo-cli/blob/86510a293529f0e862b64cf94b1eba0645bb4bce/packages/config-plugins/src/android/Properties.ts#L15 ?

Reply:
exactly. but dev-server has no dependency for config-plugins, that's why rewriting a code here.